modules/3d/include/opencv2/3d/ptcloud.hpp

```diff
@@ -11,6 +11,334 @@ namespace cv {
 //! @addtogroup _3d
 //! @{
 
+
+//! type of the robust estimation algorithm
+enum SacMethod
+{
+    /** "Random Sample Consensus: A Paradigm for Model Fitting with Applications to Image Analysis and
+     * Automated Cartography", Martin A. Fischler and Robert C. Bolles, Comm. Of the ACM 24: 381–395, June 1981.
+     */
+    SAC_METHOD_RANSAC,
+    //    SAC_METHOD_MAGSAC,
+    //    SAC_METHOD_LMEDS,
+    //    SAC_METHOD_MSAC,
+    //    SAC_METHOD_RRANSAC,
+    //    SAC_METHOD_RMSAC,
+    //    SAC_METHOD_MLESAC,
+    //    SAC_METHOD_PROSAC
+};
+
+enum SacModelType
+{
+    /** The 3D PLANE model coefficients in list [a, b, c, d],
+     corresponding to the coefficients of equation
+     ax + by + cz + d = 0. */
+    SAC_MODEL_PLANE,
+    /** The 3D SPHERE model coefficients in list [center_x, center_y, center_z, radius],
+     corresponding to the coefficients of equation
+     (x - center_x)^2 + (y - center_y)^2 + (z - center_z)^2 = radius^2.*/
+    SAC_MODEL_SPHERE,
+    //    SAC_MODEL_CYLINDER,
+
+};
+
+/**
+ *
+ * @param model_type 3D Model
+ * @return At least a few are needed to determine a model.
+ */
+inline int sacModelMinimumSampleSize(SacModelType model_type)
+{
+    switch (model_type)
+    {
+        case SAC_MODEL_PLANE:
+            return 3;
+        case SAC_MODEL_SPHERE:
+            return 4;
+        default:
+            CV_Error(cv::Error::StsNotImplemented, "SacModel Minimum Sample Size not defined!");
+    }
+}
+
+/** @brief Sample Consensus algorithm segmentation of 3D point cloud model.
+
+The following example is to use the RANSAC algorithm
+for plane segmentation of a 3D point cloud.
+
+@code{.cpp}
+using namespace cv;
+
+int planeSegmentationUsingRANSAC(const Mat &pt_cloud,
+        Mat &plane_coeffs, vector<char> &labels) {
+
+    SACSegmentation sacSegmentation;
+    sacSegmentation.setSacModelType(SAC_MODEL_PLANE);
+    sacSegmentation.setSacMethodType(SAC_METHOD_RANSAC);
+    sacSegmentation.setDistanceThreshold(0.21);
+    // The maximum number of iterations to attempt.(default 1000)
+    sacSegmentation.setMaxIterations(1500);
+    sacSegmentation.setNumberOfModelsExpected(2);
+    // Number of final resultant models obtained by segmentation.
+    int model_cnt = sacSegmentation.segment(pt_cloud,
+            labels, plane_coeffs);
+
+    return model_cnt;
+}
+@endcode
```

Comment:
Please use `@snippet` approach instead. This would guarantee that embedded code compiles.

Reply:
Modified.